NitroxServer/GameLogic/Vehicles/VehicleManager.cs

```diff
@@ -25,52 +25,65 @@ public IEnumerable<VehicleModel> GetVehicles()
 
         public void UpdateVehicle(VehicleMovementData vehicleMovement)
         {
-            if (vehiclesById.ContainsKey(vehicleMovement.Id))
+            if (vehiclesById.TryGetValue(vehicleMovement.Id, out VehicleModel vehicleModel))
             {
-                VehicleModel vehicleModel = vehiclesById[vehicleMovement.Id];
                 vehicleModel.Position = vehicleMovement.Position;
                 vehicleModel.Rotation = vehicleMovement.Rotation;
-                vehicleModel.Health = vehicleMovement.Health;
+            }
+        }
+
+        public void UpdateVehicleHealth(NitroxId vehicleId, float newHealth)
+        {
+            if (vehiclesById.TryGetValue(vehicleId, out VehicleModel vehicleModel))
+            {
+                if (newHealth > 0)
+                {
+                    vehicleModel.Health = newHealth;
+                }
+                else
+                {
+                    RemoveVehicle(vehicleId);
```

Comment:
Should we rely on the VehicleRemoved packet to do this?  In case something weird happens, the server might lose track of this instance.

Reply:
I can't think of an instance where it could happen that a vehicle has no health and is not destroyed.
Just did this to reduce the packet load (1 packet instead of 2 for the health change and vehicle kill)
There were edge cases where the non-simulating player destroyed the vehicle on his client but that's it.

Reply:
Sounds good! I didn't realize destroy packet was removed.